telemetry/specs/INT.mdk

```diff
@@ -1220,6 +1252,152 @@ Metadata if added by the source. The total stack length can be determined
+  https://github.com/p4lang/p4-applications/blob/master/telemetry/code/models/p4-dtel-metadata-semantics.yang
+  
+  Bits 0 - 14 are Baseline INT Instructions. Each instruction bit that is set
+  requests 4 bytes of metadata be sent to monitoring system at each hop, 
+  except for bits 4-6, each requires 8 bytes of metadata. 
+  
+* Domain Specific ID (16b): the unique ID of the INT Domain.
+  
+* DS Instruction (16b): Instruction bit map specific to the INT domain identified by the 
+  Domain Specific ID. Domain Specific Instruction is an instruction that requires additional 
+  processing of Domain Specific Flags (DS Flags) for the INT Domain identified by Domain Specific ID. 
+  If the Domain Specific ID matches any Domain ID known to this node, then additional processing
+  of the Domain Specific Flags and Domain Specific Instruction is required and Domain Specific
+  Metadata is also sent to the monitoring systems.
+
+  Some metadata can be defined as 'source-only' metadata by the INT domain. Those metadata
+  will be sent only by the INT source, not by INT transit or sink devices.
+  In a sense, 'source-only' bits do not serve as instructions for downstream INT devices
+  to follow. In INT-MX mode, the source node must not set the bits corresponding to 
+  "source-only" metadata in the DS Instruction bitmap. However, the source node must ensure
+  that it sets the bits corresponding to "source-only" metadata in DSMdBits bitmap of the 
+  Telemetry Report. This is needed for the monitoring system (or any consumer of the metadata) 
+  to know how to parse and use the data.
+
+  The amount of Domain Specific Metadata sent by each hop must be a multiple of 4 bytes,
+  determined from the Domain Specific Instruction. 
+  
+  If the Domain Specific ID does not match any Domain ID known to this node, then 
+  the transit node is required to either:
+  
+  - Send the metadata corresponding to INT Baseline Instruction, or
+      
+  - Skip INT processing altogether and not send any metadata the monitoring systems.
+
+* Each INT Transit device along the path that supports INT-MX sends its own metadata
+values as specified in the instruction bitmap and DS instructions.
+  - If a device is unable to provide a metadata value specified in the
+    instruction bitmap because its value is not available, it must ensure that the
+    corresponding bit in the "RepMdbit" and/or "DSMdbit" is not set in the Telemetry Report.
+  
+    Details of the metadata semantics and format of the Telemetry Report can 
+    be accessed at the link below:
+    
+    https://github.com/p4lang/p4-applications/blob/master/docs/telemetry_report_v2.0.pdf
+    
+  - Reserved bits in the instruction bitmap are to be handled as described. If an
+    INT transit hop receives a reserved bit set in the instruction bitmap (e.g.
+    set by a INT source that is running a newer version), the transit hop must
```

Comment:
the transit hop must either:
- ensure that the corresponding reserved bits in "RepMdbits" and/or "DSMdbits" 
  are not set in the Telemetry Report sent to the monitoring system, or
- for "RepMdBits", send the all-ones reserved value (4 bytes of 0xFF);
  for "DSMdbits", pad with the special all-ones reserved value in lieu of reserved
  DS Metadata, out to MD Length.

Reply:
Added.